packages/babel-plugin-transform-typescript/src/namespace.ts

```diff
@@ -73,7 +73,7 @@ function handleVariableDeclaration(
     for (const declarator of node.declarations) {
       declarator.init = t.assignmentExpression(
         "=",
-        getMemberExpression(t, name, declarator.id.name),
+        getMemberExpression(t, name, (declarator.id as t.Identifier).name),
```

Comment:
We can add `declarator is t.Identifier` assertions to the `declarator => t.isIdentifier(declarator.id)`, so TypeScript can infer that `declarator.id` here is an `t.Identifier`.

Reply:
we can do something like this:

```ts
 if (
    declarations.every((declarator): declarator is t.VariableDeclarator & {
      id: t.Identifier;
    } => t.isIdentifier(declarator.id))
  ) {
    // ...
  }
```

what do you think, does it look better?

Reply:
@zxbodya Did you push the changes?

Reply:
done, just pushed it also rebasing the changes to fix merge conflict